services/src/unit-test/java/uk/gov/moj/sdt/services/utils/SdtBulkReferenceGeneratorTest.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.moj.sdt.services.utils;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.moj.sdt.dao.GenericDao;
+import uk.gov.moj.sdt.dao.api.IGenericDao;
+import uk.gov.moj.sdt.services.utils.api.ISdtBulkReferenceGenerator;
+import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class SdtBulkReferenceGeneratorTest extends AbstractSdtUnitTestBase{
+    @Mock
+    IGenericDao genericDaoMock;
+
+    private static final String FOUR_CHARACTERS_ONLY ="The target application length is expected to be 4 characters.";
+
+    SdtBulkReferenceGenerator sdtBulkReferenceGenerator;
+    @BeforeEach
+    @Override
+    public void setUp() {
+
+        sdtBulkReferenceGenerator = new SdtBulkReferenceGenerator(genericDaoMock);
+
+    }
+
+    @Test
+    void testGetSdtBulkReferenceGenerator() {
+
+        when(genericDaoMock.getNextSequenceValue("SDT_REF_SEQ")).thenReturn(999L);
+        String generatedRef = sdtBulkReferenceGenerator.getSdtBulkReference("MCOL");
+
+        assertTrue(generatedRef.matches("^MCOL-[\\d]{14}-000000999$"), "Reference does not match expected pattern");
+        assertTrue(sdtBulkReferenceGenerator.getSdtBulkReference("MCOL").contains("999"));
+        verify(genericDaoMock,times(2)).getNextSequenceValue(anyString());
+        verify(genericDaoMock,times(2)).getNextSequenceValue("SDT_REF_SEQ");
```

Comment:
This assertTrue() can be removed as the previous one (with the regular expression) covers the same things.  This means that the ```verify(genericDaoMock,times(2)).getNextSequenceValue(anyString());``` line can be removed and the ```, times(2)``` removed from the second verify.